common/src/cams/validation.test.ts

```diff
@@ -0,0 +1,637 @@
+import V, { ValidationSpec } from './validation';
+
+describe('validation', () => {
+  describe('isString', () => {
+    const testCases = [
+      {
+        description: 'should return valid for string values',
+        value: 'hello',
+        expected: { valid: true },
+      },
+      {
+        description: 'should return invalid for number values',
+        value: 123,
+        expected: { valid: false, reason: 'Must be a string' },
+      },
+      {
+        description: 'should return invalid for boolean values',
+        value: true,
+        expected: { valid: false, reason: 'Must be a string' },
+      },
+      {
+        description: 'should return invalid for null values',
+        value: null,
+        expected: { valid: false, reason: 'Must be a string' },
+      },
+      {
+        description: 'should return invalid for undefined values',
+        value: undefined,
+        expected: { valid: false, reason: 'Must be a string' },
+      },
+      {
+        description: 'should return valid for empty string',
+        value: '',
+        expected: { valid: true },
+      },
+    ];
+    test.each(testCases)('$description', (testCase) => {
+      expect(V.isString(testCase.value)).toEqual(testCase.expected);
+    });
+  });
+  describe('minLength', () => {
+    const testCases = [
+      {
+        description: 'should return valid for string meeting minimum length',
+        minLength: 3,
```

Comment:
I would rather this test a maximally shorter word than minimum.

```suggestion
        minLength: 4,
```